src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -57,12 +63,26 @@ internal class InMemoryPartitionedAppTokenCacheAccessor : ITokenCacheAccessor
         #region Add
         public void SaveAccessToken(MsalAccessTokenCacheItem item)
         {
+            if (IsCacheOverCapacity(AccessTokenSizeInBytes))
+            {
+                _logger.Always("[AppCache] Cache is over capacity.");
+                Compact();
+            }
+
             string itemKey = item.GetKey().ToString();
             string partitionKey = CacheKeyFactory.GetClientCredentialKey(item.ClientId, item.TenantId, item.KeyId);
 
+            // Update cache size only if cache item is added, not updated
+            if (!AccessTokenCacheDictionary.TryGetValue(partitionKey, out var partition) || !partition.TryGetValue(itemKey, out _))
+            {
+                Interlocked.Add(ref _appCacheSize, AccessTokenSizeInBytes);
+                Interlocked.Add(ref TokenCache.CacheSize, AccessTokenSizeInBytes);
```

Comment:
>TokenCache.CacheSize [](http://example.com/codeflow?start=36&length=20)

Can this be calculated?

Reply:
Hmm, I don't think so. There are two instances of TokenCache class created and then one instance has app token cache accessor and one has user accessor, but they don't know about each other. And I don't think CacheSize can be moved into a higher parent class.